### PR TITLE
LPS-130893 Add  external reference code to documents

### PR DESCRIPTION
**Pull request by javierdearcos**
Added external reference code to documents services and to DocumentRepository interface and implementations.

I've deprecated old methods in Service, I've removed old methods in DocumentRepository and implementations, and I've also removed some methods in TestUtils to leave it cleaner.

### CODE REVIEW

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:test:relevant

**Comment by nhpatt** [comment]
Did Brian told you to remote these?

> com.liferay.document.library.repository.external   MAJOR      3.0.0      3.0.0      4.0.0      VERSION INCREASE REQUIRED
	>   class      com.liferay.document.library.repository.external.ExtRepositoryAdapter

If not, I would mark them as @Deprecated to avoid a direct major version change right now

File: modules/test/persistence-test/src/testIntegration/java/com/liferay/portlet/documentlibrary/service/persistence/test/DLFileEntryPersistenceTest.java [hunk 1]
```diff
@@ -129,6 +129,8 @@ public void testUpdateExisting() throws Exception {
 
 		newDLFileEntry.setUuid(RandomTestUtil.randomString());
 
+		newDLFileEntry.setExternalReferenceCode(RandomTestUtil.randomString());
```
Comment (nhpatt) [inline modules/test/persistence-test/src/testIntegration/java/com/liferay/portlet/documentlibrary/service/persistence/test/DLFileEntryPersistenceTest.java:132]: Order of commits looks strange.

File: portal-impl/src/com/liferay/portal/repository/capabilities/CapabilityRepository.java [hunk 2]
```diff
@@ -75,16 +100,17 @@ public FileEntry addFileEntry(
 
 	@Override
 	public FileEntry addFileEntry(
-			long userId, long folderId, String sourceFileName, String mimeType,
```
Comment (nhpatt) [inline portal-impl/src/com/liferay/portal/repository/capabilities/CapabilityRepository.java:78]: I think this a public API.

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java [hunk 3]
```diff
@@ -150,6 +152,138 @@ public FileEntry addFileEntry(
...
+	 *         asset category IDs, asset tag names, and expando bridge
+	 *         attributes for the file entry. In a Liferay repository, it may
+	 *         include:  <ul> <li> fileEntryTypeId - ID for a custom file entry
+	 *         type </li> <li> fieldsMap - mapping for fields associated with a
+	 *         custom file entry type </li> </ul>
+	 * @return the file entry
+	 * @throws PortalException if a portal exception occurred
+	 */
+	@Override
+	public FileEntry addFileEntry(
```
Comment (nhpatt) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java:200]: Adding these?

Comment (nhpatt) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java:200]: I see, it's github failing.

File: portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeDLFileEntry.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import java.sql.ResultSet;
+
+/**
+ * @author Javier de Arcos
+ */
+public class UpgradeDLFileEntry extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("DLFileEntry", "externalReferenceCode")) {
```
Comment (nhpatt) [inline portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeDLFileEntry.java:31]: Use the constant in the Table class here.

File: portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeDLFileEntry.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+				new AlterTableAddColumn(
+					"externalReferenceCode", "VARCHAR(75)"));
+		}
+
+		_populateExternalReferenceCode();
+	}
+
+	private void _populateExternalReferenceCode() throws Exception {
+		try (PreparedStatement ps1 = connection.prepareStatement(
+				"select fileEntryId from DLFileEntry where " +
```
Comment (nhpatt) [inline portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeDLFileEntry.java:43]: why not just one query?

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
I've checked how to make the upgrade in one query but didn't find the way and prefer to send it quickly to work in another thing.

Made the rest of changes. I've avoid breaking major where I could

**Comment by javierdearcos** [comment]
ci:forward

**Pull request closed without merging**